xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-rest/src/main/java/org/xwiki/annotation/rest/internal/AnnotationsRESTResource.java

```diff
@@ -162,39 +174,40 @@ protected AnnotationResponse postAnnotation(DocumentReference documentReference,
             // Initialize the context with the correct value.
             updateContext(documentReference);
 
-            String documentName = referenceSerializer.serialize(documentReference);
+            String documentName = this.referenceSerializer.serialize(documentReference);
 
             // check access to this function
-            if (!annotationRightService.canAddAnnotation(documentName, getXWikiUser())) {
+            if (!this.annotationRightService.canAddAnnotation(documentName, getXWikiUser())) {
                 throw new WebApplicationException(Status.UNAUTHORIZED);
             }
 
             // add the annotation
             Map<String, Object> annotationMetadata = getMap(request.getAnnotation());
-            annotationService.addAnnotation(documentName, request.getSelection(), request.getSelectionContext(),
+
+            this.annotationService.addAnnotation(documentName, request.getSelection(), request.getSelectionContext(),
                 request.getSelectionOffset(), getXWikiUser(), annotationMetadata);
             // and then return the annotated content, as specified by the annotation request
-            AnnotationResponse response = getSuccessResponseWithAnnotatedContent(documentName, request);
-            return response;
+            return getSuccessResponseWithAnnotatedContent(documentName, request);
         } catch (AnnotationServiceException | XWikiException e) {
             getLogger().error(e.getMessage(), e);
             return getErrorResponse(e);
         }
     }
 
-    /**
-     * Builds a simple map from a field collection.
-     *
-     * @param fields the collection of fields to build a map for
-     * @return a map of the fields, as string key and Object value pair
-     */
-    protected Map<String, Object> getMap(AnnotationFieldCollection fields)
+    private Map<String, Object> getMap(AnnotationFieldCollection annotation)
     {
-        Map<String, Object> metadataMap = new HashMap<String, Object>();
-        for (AnnotationField f : fields.getFields()) {
+        Map<String, Object> metadataMap = new HashMap<>();
+        for (AnnotationField f : annotation.getFields()) {
             metadataMap.put(f.getName(), f.getValue());
         }
 
+        if (metadataMap.containsKey(REQUIRES_HTML_CONVERSION)
+            && COMMENT_KEY.equals(metadataMap.get(REQUIRES_HTML_CONVERSION))) {
+            String syntax = (String) metadataMap.getOrDefault(String.format("%s_syntax", COMMENT_KEY), "xwiki/2.1");
```

Comment:
> Hard-coding the default syntax doesn't feel right.
Right.

> We should use by default (when not specified) the syntax of the current (target) document or at least the default syntax configured at wiki level.

I follow you on using the target document syntax. Not sure what you mean by "when not specified": there's a way to specify the syntax of an annotation? With a "syntax" parameter maybe? (I haven't checked yet)
Now if we rely on the target document syntax we shouldn't need a fallback on wiki level syntax, should we?

Reply:
>  there's a way to specify the syntax of an annotation?

There's no parameter specific to the annotation, but the WYSIWYG editor submits the syntax of the edited content using ``<fieldName>_syntax``, so ``comment_syntax`` in this case for instance, which the line above reads. This parameter is needed by the ``ConversionFilter`` in order to be able to perform the conversion from HTML to the specified wiki syntax. So the line above should either:

* don't use a default value (expecting the syntax to always be specified, which should be the case normally)
* or don't perform the conversion if the syntax is not specified
* or fallback on the syntax of the current document
* or fallback on the default wiki syntax

> Now if we rely on the target document syntax we shouldn't need a fallback on wiki level syntax, should we?

Of course, it's one or the other. But as I said, normally you don't even need to specify a default value for the ``comment_syntax`` parameter because the WYSIWYG editor should always send a value. This is just bullet proofing, to ensure nothing can break, but still it shouldn't be done by hard-coding the syntax.

Reply:
OK, I'll just check that the syntax is specified and don't perform conversion if that's not the case.